packages/embark-ui/src/reducers/index.js

```diff
@@ -202,13 +202,19 @@ function errorMessage(_state = null, action) {
 }
 
 function errorEntities(state = {}, action) {
-  if (!action.type.endsWith(SUCCESS)) {
+  const isSuccess = action.type.endsWith(SUCCESS);
+  if (!action.type.endsWith(FAILURE) && !isSuccess) {
     return state;
   }
   let newState = {};
   for (let name of Object.keys(entitiesDefaultState)) {
-    if (action[name] && action[name].length > 0 && action[name][0]) {
-      newState[name] = action[name][0].error;
+    if ((action.name && action.name === name) || (action[name] && action[name].length > 0 && action[name][0])) {
+      if (isSuccess) {
+        newState[name] = null;
+      } else {
+        newState[name] = action.name ? action.error : action[name][0].error;
```

Comment:
This is confusing. If the action has a name, get its error, otherwise get the last error present in the entity? Wouldn't `action[name][0].error` be the last error put in the state? What if that error was not the error from this action?

Reply:
Yeah, the last part is really confusing. Basically, it's the original error code that Anthony created. I didn't remove it because I'm not 100% sure no one uses it (though I'm 90% sure).
Maybe we should do a pass and see if maybe we can actually remove it, because it's really not an intuitive way of setting errors